data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateActionResponse.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import com.amazon.dataprepper.model.event.Event;
+
+import java.util.Optional;
+
+/**
+ * Model class to be returned in {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction}. Contains both the Event to be processed and an option
+ * to close the current window for {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateProcessor} immediately after an event is handled.
+ * @since 1.3
+ */
+public class AggregateActionResponse {
+    private Optional<Event> event;
+    private boolean closeWindowNow;
```

Comment:
I don't think we should include `closeWindow` yet. I think this approach is the right one so that we can support it, but in-case it isn't needed, let's not yet implement it.